src/physics/material/MaterialParams.cc

```diff
@@ -241,8 +271,8 @@ void MaterialParams::append_material_def(const MaterialInput& inp)
 
     ENSURE(result.number_density >= 0);
     ENSURE(result.temperature >= 0);
-    ENSURE((result.density > 0) == (inp.number_density > 0));
-    ENSURE((result.electron_density > 0) == (inp.number_density > 0));
+    ENSURE(result.density >= 0);
+    ENSURE(result.electron_density >= 0);
```

Comment:
I think these should be reverted too -- the zero cases should be "if and only if" the input is zero.